lib/data/vehicle.rb

```diff
@@ -28,7 +28,8 @@ def self.add_conditions_to_join(join_query, params)
       end
 
       def self.get_random_vehicle
-        ::Vinbot::Data::Vehicle.order(Sequel.lit('RANDOM()')).limit(1).first
+        count = ::Vinbot::Data::Vehicle.count
+        ::Vinbot::Data::Vehicle.where(id: rand(count)).first
```

Comment:
I agree with your first concern about a vehicle potentially being deleted. The RANDOM() command, however, is a very expensive task to perform. After performance testing both solutions, the original solution took about 200ms to generate a vehicle. The updated solution takes about 20ms.